### ISSUE TITLE
Unclear in documentation if SNI needs to be specified, but the postgres.js client says it should

### ISSUE THREAD

**Issue by dareiff** [neondatabase/neon#1946]
I've been working on integrating neon into a next.js app, using postgres.js, but it seems the QuickStart guides could show a few extra details if SNI is indeed required.

Specifically, I've added all of the example details, but erroring because of an unspecified project parameter.

Is it a redirect that isn't working properly?

`PostgresError: SNI is missing. EITHER please upgrade the postgres client library OR pass the project name as a parameter: '...&options=project%3D<project-name>...'.`

**Comment by yeputons** [comment]
Thank you, I will take a look into it. The following code works for me in my old Node 10. Would you mind checking if it works for you as well?

```javascript
const postgres = require('postgres');
const sql = postgres({
    host: 'frosty-sea-109598.cloud.neon.tech',
    port: 5432,
    database: 'main',
    username: 'yeputons',
    password: '[REDACTED]',
    ssl: 'require',
    connection: {
        project: 'frosty-sea-109598',
    },
});
sql`select * from ttt`.then(resp => console.log(resp)).catch(err => console.log(err));
```

The underlying issue is that we must somehow distinguish clients connecting to the same server process. We can do that easily if there is a hostname in the initial SSL handshake via SNI. Otherwise, a workaround is required. Specifying the `project` connection property is one. An even older one was to prepend project name to all usernames, IIRC, so that's probably the reason the docs are missing some bits.

**Comment by o1lab** [comment]
Running into same - connecting via TablePlus with connection string provided + SSL mode set to `required`.

**Comment by hlinnaka** [comment]
> Running into same - connecting via TablePlus with connection string provided + SSL mode set to `required`.

I found a workaround for TablePlus:

It uses libpq for the connection, so like with other programs using libpq, you can specify libpq connection parameters in the "database name" field. For us, it means that you can specify the "options=project=<project name>" like this:
![Screenshot_2022-07-07_00-58-28](https://user-images.githubusercontent.com/191602/177650575-a7b0abd4-b7a1-4f0e-9dab-271e95d8c87d.png)


So to recap, fill in the Connection dialog like this:

Database host/socket: `<project>.cloud.neon.tech`
Database's user: `<username>`
Database's name: `dbname=main options=project=<project>`

**Comment by yeputons** [comment]
We should now have much better docs about this issue at https://neon.tech/sni , I believe it covers all the necessary details
